test/end2end_test.go

```diff
@@ -6880,6 +6880,170 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
+// Implementing AuthInfo with net.Conn to send conn info up to Peer.
+type authInfoWithConn struct {
+	credentials.CommonAuthInfo
+	connection net.Conn
+}
+
+func (ai *authInfoWithConn) AuthType() string {
+	return ""
+}
+
+// connWrapperWithCloseCh wraps a net.Conn and pushes on a channel when closed.
+type connWrapperWithCloseCh struct {
+	net.Conn
+	CloseCh *testutils.Channel
```

Comment:
This one needs to unexported as well.

Reply:
mmm.. isn't this unexported right now?

Reply:
`CloseCh` field is still exported, which is immaterial given that the struct is unexported now. But it is better to not have this anomaly.

Reply:
got it. thanks. updating this